amlb/datasets/openml.py

```diff
@@ -262,8 +262,8 @@ def _get_categorical_values(self, col):
             # so we need to adapt the header accordingly.
             # Not doing so causes an issue in the R packages.
             if set(v.lower() for v in feat.nominal_values) == {"true", "false"}:
-                return [v.lower().capitalize() for v in feat.nominal_values]
-            return feat.nominal_values
+                return sorted(v.lower().capitalize() for v in feat.nominal_values)
+            return sorted(feat.nominal_values)
```

Comment:
interesting, was I doing this sorting somewhere else when rewriting the ARFF files in previous logic — meaning before changing the way those arff files are loaded / rewritten.

Reply:
I can't really find any evidence of it. It looks like the bug already existed in 97a4ac2739b446734f9aa59c5a10a244bbfd9438.

Reply:
I'm not inclined to investigate further on when it was introduced if that's okay. Sometime between the workshop paper (AutoWEKA had AUC>0.5 so it worked) and just before the data loading rework.

Reply:
I'm a bit annoyed that we have to transform the original ARFF file header for the benefit of one framework: it looks like a bug in AutoWeka imo.

Reply:
> AutoWEKA preserves the order of those variables in the column-order for their class probabilities.  And we expect those to be alphabetical.

Is not the `probabilities_labels` in `exec.py` that simply don't match the columns produced by autoweka?

Reply:
I guess it's a matter of perspective. 

 1. Auto-WEKA preserves class order as defined in the ARFF file.
 2. In the dataset meta-data we use OpenML metadata*.
 3. In Auto-WEKA exec.py there is now no longer a way to see the original order as in the ARFF file.

Makes that I don't see a way to easily set `probabilities_labels = dataset.target.values` correctly.

\* Unfortunately OpenML meta-data doesn't match the ARFF header either: [OpenML#1114](https://github.com/openml/OpenML/issues/1114).

Should #1114 be resolved s.t. the feature data matches the ARFF data I expect the bug to continue without further action.
Since without this PR the ARFF header remains unsorted, yet we also [sort values in the metadata](https://github.com/openml/automlbenchmark/blob/9e5b66d11a804b19595d602f3f3e7e54e544e201/amlb/datasets/openml.py#L106). So we would still have to change at least that.

So (I think) it came down to either keeping additional meta-data about the original ARFF header order or just sorting the header when writing to it. I went with the latter. I am not completely happy with it, but it seemed least invasive.

Reply:
At some point in the future when parquet files are the primary source of data, we have to generate the ARFF files ourselves have control over category order (except for custom ARFF files that are already split, I don't know if it would be problematic to just not support those).